### PR TITLE
Align SharedString with our factory conventions

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

SharedString now uses ISharedObjectKind and does not export the factory

## Breaking Changes

Most users of `SharedString` should be unaffected as long as they stick to the factory patterns supported by ISharedObjectKind.
If the actual class type is needed it can be found as `SharedStringClass`.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

**Review by alexvy86 — APPROVED** [review]
Devtools is fine

File: packages/framework/fluid-framework/api-report/fluid-framework.api.md [hunk 1]
```diff
@@ -516,9 +517,19 @@ export interface ISharedSegmentSequenceEvents extends ISharedObjectEvents {
 
 // @alpha
 export interface ISharedString extends SharedSegmentSequence<SharedStringSegment> {
+    annotateMarker(marker: Marker, props: PropertySet): IMergeTreeAnnotateMsg | undefined;
```
Comment (Abe27342) [inline packages/framework/fluid-framework/api-report/fluid-framework.api.md:520]: preserve the old API here: should declare `annotateMarker` as returning `void`

File: packages/dds/sequence/src/sharedString.ts [hunk 2]
```diff
@@ -165,42 +212,28 @@ export class SharedString
...
 	/**
-	 * Annotates the marker with the provided properties.
-	 * @param marker - The marker to annotate
-	 * @param props - The properties to annotate the marker with
+	 * {@inheritDoc ISharedString.annotateMarker}
 	 */
-	public annotateMarker(marker: Marker, props: PropertySet) {
-		this.guardReentrancy(() => this.client.annotateMarker(marker, props));
+	public annotateMarker(marker: Marker, props: PropertySet): IMergeTreeAnnotateMsg | undefined {
+		return this.guardReentrancy(() => this.client.annotateMarker(marker, props));
```
Comment (Abe27342) [inline packages/dds/sequence/src/sharedString.ts:232]:
```suggestion
		this.guardReentrancy(() => this.client.annotateMarker(marker, props));
```

**Review by Abe27342 — COMMENTED** [review]
looks reasonable besides the stray API change. We will also want to remove the SharedString class from the public API, but I can understand not wanting to do that here.

File: packages/dds/sequence/src/sharedString.ts (hunk 2 above)
Comment (CraigMacomber) [reply packages/dds/sequence/src/sharedString.ts:232]: ah, thats what was going on there. Good catch.

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**